src/libs/ReportUtils.ts

```diff
@@ -593,42 +778,34 @@ function findLastAccessedReport(reports, ignoreDomainRooms, policies, isFirstTim
             return sortedReports[0];
         }
 
-        return adminReport || _.find(sortedReports, (report) => !isConciergeChatReport(report));
+        return adminReport ?? sortedReports.find((report) => !isConciergeChatReport(report)) ?? null;
     }
 
     if (ignoreDomainRooms) {
         // We allow public announce rooms, admins, and announce rooms through since we bypass the default rooms beta for them.
         // Check where ReportUtils.findLastAccessedReport is called in MainDrawerNavigator.js for more context.
         // Domain rooms are now the only type of default room that are on the defaultRooms beta.
-        sortedReports = _.filter(
-            sortedReports,
-            (report) => !isDomainRoom(report) || getPolicyType(report, policies) === CONST.POLICY.TYPE.FREE || hasExpensifyGuidesEmails(lodashGet(report, ['participantAccountIDs'], [])),
+        sortedReports = sortedReports.filter(
+            (report) => !isDomainRoom(report) || getPolicyType(report, policies) === CONST.POLICY.TYPE.FREE || hasExpensifyGuidesEmails(report?.participantAccountIDs ?? []),
         );
     }
 
-    return adminReport || _.last(sortedReports);
+    return adminReport ?? sortedReports.at(-1) ?? null;
 }
 
 /**
  * Whether the provided report is an archived room
- * @param {Object} report
- * @param {Number} [report.stateNum]
- * @param {Number} [report.statusNum]
- * @returns {Boolean}
  */
-function isArchivedRoom(report) {
-    return report && report.statusNum === CONST.REPORT.STATUS.CLOSED && report.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED;
+function isArchivedRoom(report: OnyxEntry<Report> | EmptyObject): boolean {
+    return report?.statusNum === CONST.REPORT.STATUS.CLOSED && report?.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED;
 }
 
 /**
  * Checks if the current user is allowed to comment on the given report.
- * @param {Object} report
- * @param {String} [report.writeCapability]
- * @returns {Boolean}
  */
-function isAllowedToComment(report) {
+function isAllowedToComment(report: OnyxEntry<Report>): boolean {
```

Comment:
```suggestion
function isAllowedToComment(report: Report): boolean {
```
I think we should forbid null reports to be passed in this function as this may cause return true in this function.